core/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -95,7 +104,22 @@ public InternalAggregation buildAggregation(long zeroBucket) throws IOException
             final CompositeValuesSource.Collector collector =
                 array.getLeafCollector(context.ctx, getSecondPassCollector(context.subCollector));
             int docID;
+            DocIdSetIterator docIt = null;
+            if (needsScores()) {
```

Comment:
maybe reuse the local variable `needsScores` rather than call again `needsScores()`?